File: GradeTrackerAPI/Entities/Mark.cs
namespace GradeTrackerAPI.Entities
{
    public class Mark : BaseEntity
    {
        public double? Grade { get; set; }
        public string? Description { get; set; } = string.Empty;
        public double? Weighting { get; set; }
        public DateTime? Date { get; set; } = DateTime.Now;
        [ForeignKey(nameof(ModuleId))]
        public int ModuleId { get; set; }
        public virtual Module? Module { get; set; }
    }
}

File: GradeTracker/src/app/_models/mark.ts
import { CompetenceArea } from './competenceArea';
import { Module, IModule } from "./module";

export class Mark {
    id?: number;
    description?: string;
    grade?: number;
    weighting?: number;
    moduleId?: number;
    competenceArea?: CompetenceArea
}

export interface IMark {
    id?: number;
    grade?: number;
    weighting?: number;
    moduleId?: number;
}